File: python/leetcode/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        prev = None
        prev_prev = None
        while i < len(nums):
            if i > 0 and nums[i] == prev and nums[i] == prev_prev:
                i -= 1
                del nums[i]
            prev_prev = prev
            prev = nums[i]
            i+= 1
        return len(nums)

File: java/leetcode/remove-duplicates-from-sorted-array-ii.java
public class Solution {
    static int removeDuplicates(int[] nums) {
    	int l = nums.length;
    	int i = 0;
        int c = 0;
    	for (int j = 1; j < l; j++) {
            if (nums[i] != nums[j] || j + 1 == l || nums[i] != nums[j + 1]) {
                nums[++i] = nums[j];
                c = i;
            }
        }
    	return c + 1;
    }
}
